src/com/hms/flexyosisoftconnector/Tag.java

```diff
@@ -120,6 +193,417 @@ public void recordTagValue(String time)
       }
    }
 
+   /**
+    * exportBlockDescriptorHistoricalLogs
+    * Send an EBD command with a start and end time and a tag name.
+    * Start time is stored in an external file, end time is the start time
+    * plus a timeWindow. If either time is in the future, adjustments are
+    * made.
+    */
+   public void exportBlockDescriptorHistoricalLogs()
+   {
+      // Time trackers to keep track of start and end of queue
+      String startTime = TIME_ERROR;
+      String endTime = TIME_ERROR;
+
+      // Get the time tracker to use in EBD call for the start of historical logs
+      startTime = readTimeTrackerFile();
+
+      // Check that the error value did not get returned. Log if it did, and do not make EBD call
+      if (startTime.equalsIgnoreCase(TIME_ERROR)) {
+         Logger.LOG_ERR( "EBD Error: Start time not set after reading time tracker file.");
+         return;
+      }
+
+      long currentTimeMs = System.currentTimeMillis();
+      long startTimeMs = Long.parseLong(startTime);
+      int minutesConverter = 60;
+      int millisecondConverter = 1000;
+
+      // Get the time queue is behind in milliseconds and convert that to minutes
+      long howFarBehindMinutes = ((currentTimeMs - startTimeMs) / millisecondConverter / minutesConverter);
+
+      /* This comment is for the first if case
+       *
+       * There is a case we will change the time tracker to a time in the future.
+       * If we have nothing in the queue and move the tracker up by 'time window' minutes,
+       * it will make the time tracker ahead of the current time and break the EBD command.
+       *
+       * Ensure the above issue is fixed here instead of in point removal because the logic
+       * is simpler than handling when removing points.
+       *
+       * If the time period we are looking at would become a negative number... fix it.
+       */
+      if (currentTimeMs - startTimeMs < 0) {

```

Comment:
Just personal preference and future code maintainability its nice to see internal () used to encapsulate order of operations
ex. ((currentTimeMs - startTimeMs) < 0)

that way in the future if someone were to want to convert to seconds they can just do 

(((currentTimeMs - startTimeMs)/1000) < 0)

and not make a mistake like
 if (currentTimeMs - startTimeMs / 1000 < 0) {

this is more of a personal preference thing but i like that it adds explicit intent to what is being done rather than relying on the compiler.